packages/@aws-cdk/aws-lambda/lib/log-retention.ts

```diff
@@ -33,9 +34,15 @@ export interface LogRetentionProps {
  * log group. The log group is created if it doesn't already exist. The policy
  * is removed when `retentionDays` is `undefined` or equal to `Infinity`.
  */
-export class LogRetention extends cdk.Construct {
+export class LogRetention extends cdk.Construct implements logs.ILogGroup {
```

Comment:
Is this a public type? If not, I'd rename it to `LogGroupWithRetention` or something that will indicate it's a log group.

Reply:
Unfortunately it is a public type - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda/lib/index.ts#L15